src/test/java/uk/gov/companieshouse/limitedpartnershipsapi/service/LimitedPartnershipServiceTest.java

```diff
@@ -96,6 +100,29 @@ void givenTransactionAlreadyAssociatedWithAnLP_whenCreateLP_thenServiceException
         assertThrows(ServiceException.class, () -> service.createLimitedPartnership(transaction, limitedPartnershipSubmissionDto, REQUEST_ID, USER_ID));
     }
 
+    @Test
+    public void testUpdateLimitedPartnershipSubmissionWithSelfLink() {
+        //Given
+        LimitedPartnershipSubmissionDao limitedPartnershipSubmissionDao = createDao();
+        
+        // When
+        Transaction transaction = buildTransaction();
+        LimitedPartnershipSubmissionDao submission = new LimitedPartnershipSubmissionDao();
+        String submissionUri = String.format("/transactions/%s/limited-partnership/%s", transaction.getId(), limitedPartnershipSubmissionDao.getId());
+                
+        service.updateLimitedPartnershipSubmissionWithSelfLink(submission, submissionUri);
+
+        //then
+        ArgumentCaptor<LimitedPartnershipSubmissionDao> submissionCaptor = ArgumentCaptor.forClass(LimitedPartnershipSubmissionDao.class);
```

Comment:
These checks look correct but they needed to be done in the method where you had them originally. The `updateLimitedPartnershipSubmissionWithSelfLink()` method should be private.

Reply:
checks returned to original method, updateLimitedPartnershipSubmissionWithSelfLink() now private